modules/tinymce/src/themes/silver/main/ts/ui/statusbar/Statusbar.ts

```diff
@@ -75,24 +75,49 @@ const renderStatusbar = (editor: Editor, providersBackstage: UiFactoryBackstageP
 
   const getTextComponents = (): SimpleSpec[] => {
     const components: SimpleSpec[] = [];
+    const shouldRenderHelp = Options.useHelpAccessibility(editor);
+    const shouldRenderElementPath = Options.useElementPath(editor);
+    const shouldRenderRightContainer = Options.useBranding(editor) || editor.hasPlugin('wordcount');
 
-    if (Options.useElementPath(editor)) {
+    const getTextComponentClass = () => {
+      const flexStart = 'tox-statusbar__text-container--flex-start';
+      const flexEnd = 'tox-statusbar__text-container--flex-end';
+      const spaceAround = 'tox-statusbar__text-container--space-around';
+
+      if (shouldRenderHelp) {
+        const container3Columns = 'tox-statusbar__text-container-3-cols';
+
+        if (!shouldRenderRightContainer && !shouldRenderElementPath) {
+          return [ container3Columns, spaceAround ];
+        }
+
+        if (shouldRenderRightContainer && !shouldRenderElementPath) {
+          return [ container3Columns, flexEnd ];
+        }
+
+        return [ container3Columns, flexStart ];
+      }
+
+      return [ shouldRenderRightContainer && !shouldRenderElementPath ? flexEnd : flexStart ];
+    };
```

Comment:
I feel like this can be simplified, but let's make sure it works before we try to make the logic shorter :)